src/doublylinkedlist.c

```diff
@@ -106,20 +106,25 @@ DList_InsertTailList(
 void
 DList_AppendTailList(
     PDLIST_ENTRY ListHead,
-    PDLIST_ENTRY ListToAppend
+    PDLIST_ENTRY ListToBeAppendedHead
 )
 {
-    /* Codes_SRS_DLIST_06_007: [DList_AppendTailList shall place the list defined by listToAppend at the end of the list defined by the listHead parameter.] */
-    PDLIST_ENTRY ListEnd = ListHead->Blink;
+    if (!DList_IsListEmpty(ListToBeAppendedHead))
+    {
+        /* Codes_SRS_DLIST_06_007: [DList_AppendTailList shall place the list defined by listToBeAppendedHead at the end of the list defined by the listHead parameter.] */
+        PDLIST_ENTRY ListEnd = ListHead->Blink;
+
+        ListHead->Blink->Flink = ListToBeAppendedHead->Flink;
+        ListHead->Blink = ListToBeAppendedHead->Blink;
+        ListToBeAppendedHead->Blink->Flink = ListHead;
+        ListToBeAppendedHead->Flink->Blink = ListEnd;
+
+        DList_InitializeListHead(ListToBeAppendedHead);
```

Comment:
>DList_InitializeListHead [](http://example.com/codeflow?start=8&length=24)

This is calling a public API so this will be weird with regard to mocks.
What we do in these cases is that we have an internal_DList_InitializeListHead that we call from both the publicly exposed DList_InitializeListHead and from DList_AppendTailList